modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeJavaAddLayoutParameterCheck.java

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.source.formatter.check.util.JavaSourceUtil;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Tamyris Bernardo
+ */
+public class UpgradeJavaAddLayoutParameterCheck extends BaseFileCheck {
+
+	@Override
+	protected String doProcess(
+			String fileName, String absolutePath, String content)
+		throws Exception {
+
+		if (!fileName.endsWith(".java")) {
+			return content;
+		}
+
+		String newContent = content;
+
+		Matcher addLayoutMatcher = _addLayoutPattern.matcher(content);
+
+		while (addLayoutMatcher.find()) {
+			String methodCall = JavaSourceUtil.getMethodCall(
+				content, addLayoutMatcher.start());
+
+			List<String> parameterList = JavaSourceUtil.getParameterList(
+				methodCall);
+
+			if (parameterList.size() != 17) {
+				continue;
+			}
```

Comment:
All good with this pull request, @TamyrisTorres. I just have one last question to you: if tomorrow a new method with 17 parameters too is added to the classes `LayoutLocalService` and `LayoutLocalServiceUtil`, how this check will handle it? Will it be able to distinguish the 2 methods?